pipeline/transformer/transformer_mets/src/main/scala/uk/ac/wellcome/platform/transformer/mets/transformer/MetsData.scala

```diff
@@ -111,16 +111,28 @@ case class MetsData(
       ontologyType = "Work",
       value = recordIdentifier)
 
-  private def thumbnail(maybeLicense: Option[License], bnumber: String) =
-    for {
-      fileReference <- fileReferences.find(ImageUtils.isThumbnail)
-      url <- ImageUtils.buildThumbnailUrl(bnumber, fileReference)
-    } yield
-      DigitalLocation(
-        url = url,
-        locationType = LocationType("thumbnail-image"),
-        license = maybeLicense
-      )
+  private def thumbnail(bnumber: String,
+                        license: Option[License],
+                        accessStatus: Option[AccessStatus]) =
+    if (accessStatus.map(shouldCreateThumbnail).getOrElse(true))
+      fileReferences
+        .find(ImageUtils.isThumbnail)
+        .flatMap(ref => ImageUtils.buildThumbnailUrl(bnumber, ref))
+        .map { url =>
+          DigitalLocation(
+            url = url,
+            locationType = LocationType("thumbnail-image"),
+            license = license
+          )
+        } else
+      None
```

Comment:
You can write this quite nicely as:

```suggestion
    for {
      fileReference <- fileReferences.find(ImageUtils.isThumbnail)
      url <- ImageUtils.buildThumbnailUrl(bnumber, fileReference)
      if accessStatus.forall(shouldCreateThumbnail)
    } yield
      DigitalLocation(
        url = url,
        locationType = LocationType("thumbnail-image"),
        license = license
      )
```

Reply:
New GitHub feature 💯

Reply:
Yeh I prefer the other method, I just find it easier to read for some reason :laughing:. Changed to the `for` syntax as not too bothered tho

Reply:
Oh what didn't notice the `if` inside the `for` expression, thats freaky :alien: